test/dashboard/index.spec.js

```diff
@@ -25,4 +26,126 @@ describe("dashboard", () => {
     expect(dashboardWithOptions.color).to.equal("red");
     expect(dashboardWithOptions.minimal).to.be.true;
   });
+
+  describe("set* methods", () => {
+    let dashboard;
+
+    beforeEach(() => {
+      dashboard = new Dashboard();
+    });
+
+    describe("setProgress", () => {
+      it("can setProgress", () => {
+        const data = {
+          value: 0.57
+        };
+
+        expect(dashboard.setProgress(data)).to.not.throw;
+        expect(dashboard.progressbar.setProgress).to.have.been.calledOnce;
```

Comment:
I wanted to make these asserts more firm, i.e. `to.have.been.calledWith`, but I'm hitting the good ol' JS-floating-point-math-is-horrible thing. The crux of the issue is that a float like `0.57` gets evaluated in our calculations in this method as `0.56999999997` (or something of the like), and our rounding setup returns `0.56` / `56%`. Not sure if we want to update our rounding logic in this method, but wanted to call out that this is happening. I don't like the idea of writing:

```
const data = {
   value: 0.57
};

expect(dashboard.progressbar.setProgress).to.have.been.calledWith(0.56).
```